allennlp/nn/util.py

```diff
@@ -1678,3 +1678,32 @@ def find_embedding_layer(model: torch.nn.Module) -> torch.nn.Module:
                             return embedder
             return module
     raise RuntimeError("No embedding module found!")
+
+
+def extend_layer(layer: torch.nn.Module, new_dim: int, layer_type: str="linear") -> torch.nn.Module:
+    valid_layer_types = ["linear", "bilinear"]
```

Comment:
Maybe instead of this parameter you can just check `isinstance(layer, (torch.nn.Linear, torch.nn.Bilinear))`?  You're depending on specific field names in the module, so it's probably better to just check types.

Reply:
Yeah good idea - fixed that up a bit and added a test. Turns out the operation's in-place, by the way, hope that's not an issue.